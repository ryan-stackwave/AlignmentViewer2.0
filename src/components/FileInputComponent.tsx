import * as React from "react";

export interface IFileInputProps {
  labelText: string;
  onFileLoadCb?: (file: File, event: React.FormEvent) => void;
}

// Shamelessly based on from https://reactjs.org/docs/uncontrolled-components.html
export class FileInputComponent extends React.Component<IFileInputProps> {
  public static defaultProps = {
    labelText: "Upload:",
  };

  protected fileInput: React.RefObject<HTMLInputElement>;
  constructor(props: IFileInputProps) {
    super(props);
    this.handleSubmit = this.handleSubmit.bind(this);
    this.fileInput = React.createRef();
  }

  handleSubmit(event: React.FormEvent<HTMLInputElement>) {
    const { onFileLoadCb } = this.props;
    event.preventDefault();
    if (
      onFileLoadCb &&
      this.fileInput.current &&
      this.fileInput.current.files
    ) {
      onFileLoadCb(this.fileInput.current.files[0], event);
    }
    (event.target as HTMLInputElement).value = ""; //reset form
  }

  render() {
    const { labelText } = this.props;
    return (
      <label>
        <strong>{labelText}</strong>
        <input
          type="file"
          id="files"
          ref={this.fileInput}
          onInput={this.handleSubmit}
        />
      </label>
    );
  }
}
