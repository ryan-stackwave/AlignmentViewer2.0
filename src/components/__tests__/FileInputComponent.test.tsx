import "jest-webgl-canvas-mock";
import * as React from "react";

import { shallow } from "enzyme";

import { FileInputComponent } from "../FileInputComponent";

// Due to the runtime necessities of using styles, we need to explicitly mock out some stub data.
// https://github.com/facebook/jest/issues/3094
jest.mock("../MolecularStyles.module.scss", () => {
  return {
    aaStyBGAlpha_Default: 0,
    aaStyClass_Default: "mock-aa-class",
    aaStyColorOrder_Default: "",
    aaStyColors_Default: "",
    aaStyDesc_Default: "mock-aa-style-desc",
    ntStyBGAlpha_Default: "",
    ntStyClass_Default: "mock-nt-class",
    ntStyColorOrder_Default: "",
    ntStyColors_Default: "",
    ntStyDesc_Default: "mock-nt-style-desc",
  };
});

describe("FileInputComponent", () => {
  it("Should render with no props.", () => {
    const wrapper = shallow(<FileInputComponent />);
    expect(wrapper).toMatchSnapshot();
  });
});
