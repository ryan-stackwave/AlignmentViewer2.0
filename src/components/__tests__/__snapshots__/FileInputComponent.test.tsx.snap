// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileInputComponent Should render with no props. 1`] = `
<label>
  <strong>
    Upload:
  </strong>
  <input
    id="files"
    onInput={[Function]}
    type="file"
  />
</label>
`;
